### PR TITLE
Remove data layer for Directly, do the requests directly

### PR DESCRIPTION
**Pull request by jsnajdr**
Removes the data layer that was wrapping the `Directly.initialize` and `Directly.askQuestion` functions and converts the action to thunks instead. That will allow us to remove the dependency on data layer code (middleware etc) from the Inline Help widget, making it smaller.

**How to test:**
Open the inline help popup, go to a free site that is supported only in public forums, and ask a question there. In development mode, don't worry about posting the question to a real site: there is a sandbox configured with the `config( 'directly_rtm_widget_environment' )` variable. Verify that it all works.

### CODE REVIEW

**Comment by jsnajdr** [comment]
Here are more detailed test instructions. To simulate the Directly widget on your typical a11n account, you'll need to manually disable all other kinds of support: happychat, kayako, ... To achieve that, you need to modify the `getSupportVariation` selector to always return `SUPPORT_DIRECTLY` instead of checking the various `isHappychatUserEligible` or `isTicketSupportEligible` flags.

Then the Inline Help popup should show an "Ask an Expert" UI:

Screenshot 2021-11-09 at 10 50 17

and after posting the question, a iframe widget with an external HTML/script should show up:

Screenshot 2021-11-09 at 10 51 03

@tyxla could you have a look at this one in near future?

**Comment by tyxla** [comment]
> Here are more detailed test instructions. To simulate the Directly widget on your typical a11n account, you'll need to manually disable all other kinds of support: happychat, kayako, ... To achieve that, you need to modify the `getSupportVariation` selector to always return `SUPPORT_DIRECTLY` instead of checking the various `isHappychatUserEligible` or `isTicketSupportEligible` flags.

I've done that (returning `SUPPORT_DIRECTLY` in the `getSupportVariation` selector) but have been unable to get it to work - I get only an infinitely loading popup:



Any idea what I might be missing when testing this?

**Comment by jsnajdr** [comment]
I think you'll need to fake the `isSupportVariationDetermined` selector, too. When its "not determined yet", you get a placeholder. Always return `true` and you should be set?

> To achieve that, you need to modify the `getSupportVariation` selector to always return `SUPPORT_DIRECTLY`

To be honest, what I _really_ did during my own testing was a bit different and more complex 🙂  I patched multiple places, like the `isHappychatUserEligible`, `isTicketSupportEligible` or `isDirectlyReady` selectors.

**Review by tyxla — COMMENTED** [review]
I've managed to kinda test it by altering:

* `getSupportVariation()` to return `SUPPORT_DIRECTLY`
* `getDirectlyStatus()` to return `true`
* `isSupportVariationDetermined()` to return `true`.

However, when posting the question, I'm getting this error:



So I guess it's not enough to test it properly. 

I'd recommend getting someone familiar with Directly or at least with our happiness tooling to test this thoroughly.

In the meantime, I'm currently doing a code review so I can be of help with that at least.

File: client/state/help/directly/reducer.js [hunk 1]
```diff
@@ -9,10 +9,8 @@ import { STATUS_UNINITIALIZED, STATUS_INITIALIZING, STATUS_READY, STATUS_ERROR }
 
 export const questionAsked = ( state = null, action ) => {
 	switch ( action.type ) {
-		case DIRECTLY_ASK_QUESTION: {
-			const { questionText, name, email } = action;
-			return { questionText, name, email };
-		}
+		case DIRECTLY_ASK_QUESTION:
+			return true;
```
Comment (tyxla) [inline client/state/help/directly/reducer.js:13]: This change makes sense, we don't care about the question details, but only whether it's been asked or not.

File: client/state/help/directly/actions.js [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
 
-export function askQuestion( questionText, name, email ) {
-	return { type: DIRECTLY_ASK_QUESTION, questionText, name, email };
-}
-
-export function initialize() {
-	return { type: DIRECTLY_INITIALIZATION_START };
-}
+export const askQuestion = ( questionText, name, email ) => async ( dispatch ) => {
+	dispatch( { type: DIRECTLY_ASK_QUESTION } );
```
Comment (tyxla) [inline client/state/help/directly/actions.js:12]: Seems like we're optimistically setting the state to be in a "asked question" state, while it's not clear that `directly.askQuestion()` will be successful and if the question will actually be asked successfully. But it was like that before, so 🤷

**Review by tyxla — COMMENTED** [review]
Code-wise this is looking great! I couldn't spot any regressions.

However, I'd recommend someone familiar with the happiness tooling and flows to give it a thorough testing before we ship it.

**Review by mattwondra — APPROVED** [review]
Rather than altering Calypso code to force Directly to load, you should sign in as a free user. Only a portion of free users are offered Directly, so you'll also need to sandbox the WPCOM API and return a success state from the top of `/public.api/rest/wpcom-json-endpoints/class.wpcom-json-api-help-directly-endpoint.php`:

```
return [ 'isAvailable' => true ];
```

This should force that user into Directly without manipulating Calypso code at all so you can test the flow accurately. I've tested this way and Directly requests are being correctly sent still.

Also @jsnajdr I'm no longer on HappOps, you can ping @Automattic/happops-devs for future requests for review on Happiness-related code. :+1:

**Comment by jsnajdr** [comment]
> However, when posting the question, I'm getting this error:

To get around this, you just need to fake the response of the `/help/directly/mine` endpoint here:

https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/directly/index.js#L116

Normally, only a certain percentage of users is sent to Directly, assigned by user ID: `( user_id % 100 ) < percent_threshold`

> I'd recommend getting someone familiar with Directly or at least with our happiness tooling to test this thoroughly.

I managed to verify it myself and posted some screenshots as evidence, so I'm fairly confident that it's OK. But anyway, @mattwondra, you're probably the only person familiar with this, could you have a quick look?

**Comment by jsnajdr** [comment]
> Normally, only a certain percentage of users is sent to Directly

This is currently set at 25%:

Screenshot 2021-11-10 at 13 26 52

WP.com Network Admin has a screen for setting that, in the Happiness -> Live Chat Options menu.

**Comment by jsnajdr** [comment]
Thank you @mattwondra, and I apologize for distracting you!

**Pull request merged**